Fix the config.php owner check for console.php

phlb opened the pull request:
The config directory must be taken from OC::$configDir

The config directory was hardcoded in console.php.

phlb committed:
Fix the config.php owner check for console.php

The config directory must be taken from OC::$configDir

Xenopathic commented on console.php:
```diff
@@ -31,7 +31,7 @@
 		}
 		$user = posix_getpwuid(posix_getuid());
-		$configUser = posix_getpwuid(fileowner(OC::$SERVERROOT . '/config/config.php'));
+		$configUser = posix_getpwuid(fileowner(OC::$configDir . "/config.php" ));
```
Please use single quotes for the literal string part. Also, `$configDir` already has the trailing slash, so you might need to remove that from the literal string.

MorrisJobke commented:
cc @DeepDiver1975

DeepDiver1975 commented:
This change has not one single benefit - configDir is always(within the scope of this script) pointing to SERVERROOT . '/config'

Xenopathic commented:
@DeepDiver1975 But on the other hand, it does make the intent a little more obvious, and pre-empts any possible-potential bugs we might at some point conceivably-prospectively have. Yes, it's unlikely we will break this, but the possibility is still there :smile:

DeepDiver1975 commented:
I'm willing to accept this change if @phlb will respond to the comments above

phlb committed:
Minor fix

 - Replace double quote by simple quote
 - Remove uneeded slash in the configdir path.

phlb commented:
Ok for the modification.

@DeepDiver1975 the configdir depends on lib/base.php. In several cases, the config dir is not ``OC::$SERVERROOT . '/config/'``. Currently only for testing purposes but also for the PR #16424.

Xenopathic commented:
:+1:

MorrisJobke commented:
:+1:

MorrisJobke commented:
I will port this to directly to master and close this PR, because it is against a stable version.

MorrisJobke commented:
#17546
